realm/realm-library/src/main/java/io/realm/internal/async/RealmThreadPoolExecutor.java

```diff
@@ -38,6 +42,7 @@
     private boolean isPaused;
     private ReentrantLock pauseLock = new ReentrantLock();
     private Condition unpaused = pauseLock.newCondition();
+    private List<Future<?>> transactions = Collections.synchronizedList(new ArrayList<Future<?>>());
```

Comment:
I still don't think use a list `transactions` is a good idea. The major problems right now:
1. It costs memory which is not needed
2. There is a leak until `Realm.close()` gets called. This list will grows without being cleaned up.

And also i am totally fine we remove the warning part out of the scope of this PR, instead of focusing on the shared group version part.